src/client/visualizations/base-visualization/base-visualization.tsx

```diff
@@ -54,133 +35,18 @@ export class BaseVisualization<S extends BaseVisualizationState> extends React.C
-    return this.debouncedCallExecutor(essence, timekeeper);
-  }
-
-  protected getQuery(essence: Essence, timekeeper: Timekeeper): Expression {
-    return makeQuery(essence, timekeeper);
-  }
-
-  private callExecutor = (essence: Essence, timekeeper: Timekeeper): Promise<DatasetLoad | null> =>
-    essence.dataCube.executor(this.getQuery(essence, timekeeper), { timezone: essence.timezone })
-      .then((dataset: Dataset) => {
-          // signal out of order requests with null
-          if (!this.wasUsedForLastQuery(essence)) return null;
-          return loaded(dataset);
-        },
-        err => {
-          // signal out of order requests with null
-          if (!this.wasUsedForLastQuery(essence)) return null;
-          reportError(err);
-          return error(err);
-        });
-
-  private wasUsedForLastQuery(essence: Essence) {
-    return essence.equals(this.lastQueryEssence);
-  }
-
-  private debouncedCallExecutor = debounceWithPromise(this.callExecutor, 500);
-
-  private handleDatasetLoad(dl: DatasetLoad, derivedState: Partial<S> = {}) {
-    // as object will be fixed in typescript 3.2 https://github.com/Microsoft/TypeScript/issues/10727
-    this.setState({ ...(derivedState as object), datasetLoad: dl });
-    const { registerDownloadableDataset } = this.props;
-    if (registerDownloadableDataset) {
-      registerDownloadableDataset(isLoaded(dl) ? dl.dataset : null);
-    }
-  }
-
-  protected shouldFetchData(nextProps: VisualizationProps): boolean {
-    return this.differentVisualizationDefinition(nextProps);
-  }
-
-  private differentVisualizationDefinition(nextProps: VisualizationProps) {
-    const { essence, timekeeper } = this.props;
-    const nextEssence = nextProps.essence;
-    const nextTimekeeper = nextProps.timekeeper;
-    return nextEssence.differentDataCube(essence) ||
-      nextEssence.differentEffectiveFilter(essence, timekeeper, nextTimekeeper) ||
-      nextEssence.differentTimeShift(essence) ||
-      nextEssence.differentSplits(essence) ||
-      nextEssence.differentSeries(essence) ||
-      nextEssence.differentSettings(essence) ||
-      this.differentBucketingTimezone(nextEssence) ||
-      this.differentLastRefreshRequestTimestamp(nextProps);
-  }
-
-  private differentBucketingTimezone(newEssence: Essence): boolean {
-    const { essence } = this.props;
-    return !essence.timezone.equals(newEssence.timezone) && newEssence.splits.hasSplitOn(essence.getTimeDimension());
-  }
-
-  private differentLastRefreshRequestTimestamp({ refreshRequestTimestamp }: VisualizationProps): boolean {
-    return refreshRequestTimestamp !== this.props.refreshRequestTimestamp;
-  }
-
-  private visualisationNotResized(nextProps: VisualizationProps): boolean {
-    return this.props.stage.equals(nextProps.stage);
+    return {};
   }
 
   protected renderInternals(dataset: Dataset): JSX.Element {
     return null;
   }
 
-  deriveDatasetState(dataset: Dataset): Partial<S> {
```

Comment:
Still need to resolve this. Probably Visualizations need to memoize upon receiving data prop